src/text/word2vec-utils.ts

```diff
@@ -0,0 +1,90 @@
+export function clean(text: string, stopwords: string[] | undefined): Array<string> {
+  let textArray = text.split(' ');
+  //filter out empty string
+  textArray = textArray.filter((value) => {
+    return value != '';
+  });
+
+  //filter out stopwords.
+  if (stopwords) {
+    textArray = textArray.filter((value) => {
+      return !stopwords.includes(value);
+    });
+  }
+  //remove punctuation
+  textArray = textArray.map((value) => {
+    return value.replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g, "");
```

Comment:
```suggestion
    return value.replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g, '');
```

And should we think about other languages like Chinese?

Reply:
Yes, we should.

Reply:
May I ask you to add a TODO comment, thus I will add the corresponding rules for Chinese language :p